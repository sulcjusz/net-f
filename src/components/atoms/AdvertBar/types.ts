export interface IAdvertBarProps {
  children: React.ReactNode
  className?: string
}
