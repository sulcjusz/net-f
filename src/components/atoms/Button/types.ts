export interface IButtonProps {
  onClick?: () => void
  size?: 'small' | 'medium'
  isActive?: boolean
  children: React.ReactNode
  className?: string
}
