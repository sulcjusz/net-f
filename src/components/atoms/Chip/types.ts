export interface IChipProps {
  size?: 'small' | 'medium'
  textColor?: 'black' | 'white'
  children: React.ReactNode
  className?: string
}
