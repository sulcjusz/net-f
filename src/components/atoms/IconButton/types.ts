export interface IIconButtonProps {
  onClick?: () => void
  size?: 'small' | 'medium'
  disabled?: boolean
  icon: React.ReactNode
  className?: string
}
