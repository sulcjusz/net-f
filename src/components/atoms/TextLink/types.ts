export interface ITextLinkProps {
  text: string
  onClick: () => void
  className?: string
}
