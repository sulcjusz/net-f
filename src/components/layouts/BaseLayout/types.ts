export interface IBaseLayoutProps {
  children: React.ReactNode
}
