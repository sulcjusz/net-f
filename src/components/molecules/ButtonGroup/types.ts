export interface IButtonGroupProps {
  children: React.ReactNode
}
