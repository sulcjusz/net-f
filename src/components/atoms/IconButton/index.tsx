import React from 'react'
import classnames from 'classnames'
import {IIconButtonProps} from './types'

const IconButton: React.FC<IIconButtonProps> = ({
  onClick,
  isActive = false,
  size = 'medium',
  icon,
  className
}) => {
  const buttonClasses = classnames(
    'rounded-full border border-grey-medium font-exo leading-5 tracking-[.07em] text-center',
    {
      'text-grey-400 bg-grey-200 font-semibold': isActive,
      'text-grey-400 bg-grey-300 font-medium hover:bg-opacity-50 hover:border-grey-medium':
        !isActive,
      'text-base py-2 px-2': size === 'small',
      'text-xl py-4 px-4': size === 'medium',
      'pointer-events-none': isActive
    },
    className
  )

  return (
    <button className={buttonClasses} onClick={onClick}>
      {icon}
    </button>
  )
}

export default IconButton
