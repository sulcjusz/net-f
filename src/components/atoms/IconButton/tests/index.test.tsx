import {render, fireEvent, RenderResult} from '@testing-library/react'
import '@testing-library/jest-dom'

import IconButton from '../index'
import {IIconButtonProps} from '../types'

const renderIconButton = ({
  onClick,
  disabled = false,
  size = 'medium',
  icon,
  className
}: IIconButtonProps): {button: HTMLElement} & RenderResult => {
  const utils = render(
    <IconButton
      onClick={onClick}
      disabled={disabled}
      size={size}
      icon={icon}
      className={className}
    />
  )
  const button = utils.getByRole('button')
  return {...utils, button}
}

describe('IconButton component', () => {
  it('should match the snapshot', () => {
    const {button} = renderIconButton({
      icon: <svg />
    })
    expect(button).toMatchSnapshot()
  })

  it('should render icon button correctly', () => {
    const {button} = renderIconButton({
      icon: <svg />
    })
    expect(button).toBeInTheDocument()
    expect(button.tagName.toLowerCase()).toBe('button')
    expect(button).toHaveClass(
      'rounded-full border border-grey-medium font-exo leading-5 tracking-[.07em] text-center text-xl p-4'
    )
  })

  it('should apply custom class name', () => {
    const {button} = renderIconButton({
      icon: <svg />,
      className: 'custom-class'
    })
    expect(button).toHaveClass('custom-class')
  })

  it('should trigger onClick event', () => {
    const onClickMock = jest.fn()
    const {button} = renderIconButton({
      icon: <svg />,
      onClick: onClickMock
    })
    fireEvent.click(button)
    expect(onClickMock).toHaveBeenCalledTimes(1)
  })

  it('should render icon with isActive style when isActive is true', () => {
    const {button} = renderIconButton({
      icon: <svg />,
      disabled: true
    })
    expect(button).toHaveClass('text-grey-400 bg-grey-200 font-semibold')
  })

  it('should render icon with default style when isActive is false', () => {
    const {button} = renderIconButton({
      icon: <svg />,
      disabled: false
    })
    expect(button).toHaveClass(
      'text-grey-400 bg-grey-300 font-medium hover:bg-opacity-50 hover:border-grey-medium'
    )
  })

  it('should render icon with disabled style when disabled is true', () => {
    const {button} = renderIconButton({
      icon: <svg />,
      disabled: true
    })
    expect(button).toHaveClass('text-grey-400 bg-grey-200 font-semibold')
  })

  it('should render small size icon button', () => {
    const {button} = renderIconButton({
      icon: <svg />,
      size: 'small'
    })
    expect(button).toHaveClass('text-base p-2')
  })

  it('should render medium size icon button', () => {
    const {button} = renderIconButton({
      icon: <svg />,
      size: 'medium'
    })
    expect(button).toHaveClass('text-xl p-4')
  })
})
