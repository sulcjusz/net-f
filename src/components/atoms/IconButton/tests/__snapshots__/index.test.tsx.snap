// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton component should match the snapshot 1`] = `
<button
  class="rounded-full border border-grey-medium font-exo leading-5 tracking-[.07em] text-center text-grey-400 bg-grey-300 font-medium hover:bg-opacity-50 hover:border-grey-medium text-xl p-4"
>
  <svg />
</button>
`;
